### PR TITLE
Test to validate the EC2Cloud private key migration 

### PR DESCRIPTION
**Pull request by varyvol**
Adds to #498 an automated test validating the migration for an EC2 cloud is correctly done.

@MRamonLeon @djesionek

### CODE REVIEW

File: src/test/java/hudson/plugins/ec2/EC2CloudMigrationTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import static org.junit.Assert.*;
+
+public class EC2CloudMigrationTest {
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @Test
+    @LocalData
```
Comment (MRamonLeon) [inline src/test/java/hudson/plugins/ec2/EC2CloudMigrationTest.java:21]: Could you add some comment explaining that the local data is from a previous version of EC2 to show it clear the migration works? Or maybe in the config.xml you can add a comment noticing the `privateKey` is the actual key and not a credential and it works.

**Review by MRamonLeon — COMMENTED** [review]
I guess this PR superseeds https://github.com/jenkinsci/ec2-plugin/pull/498 right?

File: src/test/java/hudson/plugins/ec2/EC2CloudMigrationTest.java (hunk 1 above)
Comment (varyvol) [reply src/test/java/hudson/plugins/ec2/EC2CloudMigrationTest.java:21]: Added.

**Comment by varyvol** [comment]
I don't pretend this to supersede the original PR, only to demonstrate how to use LocalData to perform the related testing.

**Review by MRamonLeon — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by CptPlastic** [comment]
This breaks the plugin. We ended up rolling back.

**Comment by varyvol** [comment]
@CptPlastic this only adds a test, so I guess you'll be referring to a previous PR. Which version did you update to?

**Comment by timja** [comment]
this is raised as https://issues.jenkins-ci.org/browse/JENKINS-63707

**Pull request merged**